src/main/java/com/acrolinx/client/sdk/internal/SignInPollResponse.java

```diff
@@ -18,13 +18,9 @@ private SignInPollResponse()
     {
     }
 
-    static final TypeSelector<SignInPollResponse> TYPE_SELECTOR = new TypeSelector<SignInPollResponse>() {
-        @Override
-        public Class<? extends SignInPollResponse> getClassForElement(JsonElement readElement)
-        {
-            Boolean isSuccess = readElement.getAsJsonObject().has("data");
-            return isSuccess ? SignInPollResponse.Success.class : Progress.class;
-        }
+    static final TypeSelector<SignInPollResponse> TYPE_SELECTOR = readElement -> {
```

Comment:
Lambdas over anonymous classes for readability and reducing boilerplate code. Let the compiler work for you 🙃.